Add NFT Auction topic (#1735)
XRPLF/xrpl-dev-portal: content/concepts/tokens/nftoken-auctions.md

---
html: nftoken-auctions.html
parent: non-fungible-tokens.html
blurb: You can assign another account to mint NFTs in your stead.
labels:
 - Non-fungible Tokens, NFTs
---

# Running an NFT Auction

There are several ways to run an auction, each with advantages and disadvantages.

> ## Run the Auction Off the XRPL, Complete the Purchase on XRPL

Reviewer: ```suggestion
## Run the Auction and Complete the Purchase on the XRPL

Author: I understand your reasoning, but the main point of this option is that the auction is not facilitated by the XRPL.

> This flow is listed first as it is the most straightforward. You store your bids in a private database. You take a cut of the winning bid prior to sending the Buyer/Seller the XRPL transaction that completes the purchase. Note that because NFTokenOffer objects can always be canceled by their creator, it is not currently possible to implement a binding offer.

Reviewer: ```suggestion
This flow is the most straightforward. Note that the `NFTokenOffer` objects can always be canceled by their creator, so it's not possible to implement a binding offer.

1. Store your bids in a private database.
2. You take a cut of the winning bid.
3. Send the buyer/seller the XRPL transaction to complete the purchase. 
```
I thought our style guide was to use contractions? If not, I need to make some adjustments to my docs moving forward.

Author: 35 years of leaving contractions out of developer documentation is a hard habit to break, particularly since I don't agree. Our style guide does say we can use common contractions, but Tina will update the guide to avoid some of the more awkward ones.

## Run the Auction in Brokered Mode, with a Reserve

Run the auction in brokered mode, as an auction with a reserve.

![Auction in Brokered Mode with a Reserve](img/nft-auction1.png "Auction in Brokered Mode with a Reserve")

> 1. The Seller creates the NFT, then sets the auction reserve price using NFTokenCreateOffer, specifying the broker account as the destination.

Reviewer: ```suggestion
1. The Seller creates the NFT, then sets the auction reserve price using `NFTokenCreateOffer`, specifying the broker account as the destination.
```
I'm noticing Buyer and Seller are sometimes capitalized. Suggest sticking to one.

> 1. Buyers make bids using NFTokenCreateOffer, setting the broker account as the destination.

Reviewer: ```suggestion
1. Buyers make bids using `NFTokenCreateOffer`, setting the broker account as the destination.
```

> 1. The Broker selects the winning bid, completes the sale using NFTokenAcceptOffer, collecting the broker fee. Then the Broker cancels any losing bids using NFTokenCancelOffer.

Reviewer: ```suggestion
1. The Broker selects the winning bid, completes the sale using `NFTokenAcceptOffer`, collecting the broker fee. Then the Broker cancels any losing bids using `NFTokenCancelOffer`.
```

**Pros:**

> - The entire auction happens on the XRPL, including your broker fee.

Reviewer: Maybe I'm misreading the topic, but aren't all three methods listed taking place entirely on the XRPL?

Author: No, the first option handles the _auction_ independent of the XRPL, completing the sale as a normal transaction. Options 2 and 3 use the broker process to handle the auction using the brokered mode feature.

- The seller represents their reserve price on-chain.
- This is _close_ to a binding offer, from the buyside.

**Cons:**

- There must be implicit trust between the seller and the marketplace that the marketplace will not take more than some previously-agreed-upon rate. If the reserve was 1 XRP and the winning bid was 1000 XRP, there is no on-chain mechanism to prevent the broker from taking 999 XRP as profit, leaving only the reserve profits for the Seller.

> A major mitigating factor of this downside is that if this behavior were to happen, Brokers would lose their entire market share, which sellers Should understand.

Reviewer: ```suggestion
A major mitigating factor of this downside is that if this behavior were to happen, Brokers would lose their entire market share, which sellers should understand.
```

## Run the Auction in Brokered Mode, without a Reserve. 

This is the most complex workflow of the three.

![Auction in Brokered Mode without a Reserve](img/nft-auction2.png "Auction in Brokered Mode without a Reserve")

> 1. The seller creates an NFT using NFTokenMint.

Reviewer: ```suggestion
1. The seller creates an NFT using `NFTokenMint`.
```

> 1. The bidders make bids using NFTokenCreateOffer, setting the broker as the destination.

Reviewer: ```suggestion
1. The bidders make bids using `NFTokenCreateOffer`, setting the broker as the destination.
```

> 1. The broker selects the winning bid, subtracts the amount to be collected as a fee, then requests theseller sign a sell off for this amount via NFTokenCreateOffer.

Reviewer: ```suggestion
1. The broker selects the winning bid, subtracts the amount to be collected as a fee, then requests the seller sign a sell offer for this amount via `NFTokenCreateOffer`.
```

1. The seller signs the requested offer, setting the broker as the destination.

> 1. The broker completes the sale using NFTokenAcceptOffer, and receives the broker fee.

Reviewer: ```suggestion
1. The broker completes the sale using `NFTokenAcceptOffer`, and receives the broker fee.
```

> 1. The broker cancels any remaining bids using NFTokenCancelOffer.

Reviewer: ```suggestion
1. The broker cancels any remaining bids using `NFTokenCancelOffer`.
```

**Pros:**

> - This flow requires absolutely no trust among participants, making it option one would most often expect on the blockchain.

Reviewer: ```suggestion
- This flow requires absolutely no trust among participants, making it the option most people expect on the blockchain.
```

Author: I'll take this one, but I actually prefer the original. The edit is more conversational.

> - Sellers know exactly how much the marketplace take from them in fees and must agree to it on chain.

Reviewer: ```suggestion
- Sellers know exactly how much the marketplace takes from them in fees and must agree to it on the chain.
```

**Cons:**

- After the auction is complete, the sale is contingent on the seller agreeing to the final bid amount and broker fee amount. This means that sellers can back out of an otherwise complete auction or that sellers can delay settlement due to being distracted or not seeing some notification.

> - After the auction is complete, a seller can actively refuse the winning bid, instead selling to someone else.

Reviewer: ```suggestion
- After the auction is complete, a seller can refuse the winning bid, instead selling to someone else.
```

---
html: nftoken-auctions.html
parent: non-fungible-tokens.html
blurb: You can assign another account to mint NFTs in your stead.
labels:
 - Non-fungible Tokens, NFTs
---

# Running an NFT Auction

There are several ways to run an auction, each with advantages and disadvantages.

## Run the Auction Off the XRPL, Complete the Purchase on XRPL

This flow is the most straightforward. Note that the `NFTokenOffer` objects can always be canceled by their creator, so it's not possible to implement a binding offer.

> 1. Store your bids in a private database.

Reviewer 2: just curious if second person phrasing is intentional here? The assumption is that "you" are the broker? I think the rest of the documentation uses third person

Author: You're correct. Typically, we write in second person. I made the mistake because I took the text from the flowchart. On reflection, I think the second person works well in the first scenario, third person in the second and third scenarios. I'm going to leave it for now.

2. You take a cut of the winning bid.
3. Send the buyer/seller the XRPL transaction to complete the purchase. 

## Run the Auction in Brokered Mode, with a Reserve

Run the auction in brokered mode, as an auction with a reserve.

![Auction in Brokered Mode with a Reserve](img/nft-auction1.png "Auction in Brokered Mode with a Reserve")

1. The Seller creates the NFT, then sets the auction reserve price using `NFTokenCreateOffer`, specifying the broker account as the destination.

> 1. Buyers make bids using `NFTokenCreateOffer`, setting the broker account as the destination.

Reviewer 3: nit: would this be considered Bidders instead of Buyers as they have not necessarily won the bid yet?

Reviewer 2: on line 50 it does says `Bidders`, so i think it is meant to say it here as well

Author: Yes, updated.

Author: Changed "Buyers make bids" to "Bidders make offers," which is more accurate.

1. The Broker selects the winning bid, completes the sale using `NFTokenAcceptOffer`, collecting the broker fee. Then the Broker cancels any losing bids using `NFTokenCancelOffer`.

**Pros:**

- The entire auction happens on the XRPL, including your broker fee.
- The seller represents their reserve price on-chain.
- This is _close_ to a binding offer, from the buyside.

**Cons:**

> - There must be implicit trust between the seller and the marketplace that the marketplace will not take more than some previously-agreed-upon rate. If the reserve was 1 XRP and the winning bid was 1000 XRP, there is no on-chain mechanism to prevent the broker from taking 999 XRP as profit, leaving only the reserve profits for the Seller.

Reviewer 3: nit: I'm noticing some inconsistencies on the capitalization for "Buyer", "Seller", and "Broker", do we intentionally want to have different variants based on sentence context?

Reviewer 2: maybe its good to clarify that marketplace = broker?
